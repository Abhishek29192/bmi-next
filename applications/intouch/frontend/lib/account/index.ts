import axios from "axios";
import { v4 } from "uuid";

export const mutationCreateAccount = `mutation CreateAccount($input: CreateAccountInput!) {
  createAccount(input: $input) {
    account {
      id
      marketId
    }
  }
}`;
export const createAccount = async (req, session) => {
  const { AUTH0_NAMESPACE } = process.env;
  const { user } = session;

  // const market = user[`${AUTH0_NAMESPACE}/market`];
  const firstName = user[`${AUTH0_NAMESPACE}/firstname`];
  const lastName = user[`${AUTH0_NAMESPACE}/lastname`];
  const type = user[`${AUTH0_NAMESPACE}/type`];
  const market = user[`${AUTH0_NAMESPACE}/market`];

  const body = {
    query: mutationCreateAccount,
    variables: {
      input: {
        firstName,
        lastName,
        email: user.email,
        role: type === "company" ? "COMPANY_ADMIN" : "INSTALLER",
        marketCode: market
      }
    }
  };

  try {
    const { data } = await axios.post(
      `http://${req.headers.host}/api/graphql`,
      body,
      {
        headers: {
          authorization: `Bearer ${session.accessToken}`,
          "x-request-id": v4()
        }
      }
    );

    return data;
  } catch (error) {
    // eslint-disable-next-line no-console
    console.log("Error:message: ", error.message);
  }
};
