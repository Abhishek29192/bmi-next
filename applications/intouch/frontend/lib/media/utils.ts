import { GetMediaFoldersQuery } from "../../graphql/generated/operations";
import { MediaItem, MediaFolders, RootFolders, MediaTool } from "./types";

export const getRootFolders = ({
  marketContentCollection
}: GetMediaFoldersQuery): RootFolders => {
  return marketContentCollection.items[0]?.mediaLibraryRootCollection?.items;
};

export const findFolderById = (allFolders: MediaFolders, folderId: string) => {
  return allFolders.find((f) => f.sys.id === folderId);
};

export const getParentFolder = (mediaItem: MediaItem, folders: MediaFolders) =>
  folders.find((f) =>
    f.childrenCollection.items.find(
      (child) => child.sys.id === mediaItem.sys.id
    )
  );

export const getMediaItemPath = (
  mediaItem: MediaItem,
  allFolders: MediaFolders,
  rootFolders: RootFolders,
  currentPath: MediaItem[] = []
) => {
  if (!mediaItem) {
    return [];
  }
  const rootFolder = findFolderById(rootFolders, mediaItem.sys.id);

  if (rootFolder) {
    return [rootFolder, ...currentPath];
  }
  return getMediaItemPath(
    getParentFolder(mediaItem, allFolders),
    allFolders,
    rootFolders,
    [mediaItem, ...currentPath]
  );
};

const META_TYPES = {
  FOLDER: "FOLDER",
  IMAGE: "IMAGE",
  VIDEO: "VIDEO",
  EXTERNAL_LINK: "EXTERNAL_LINK",
  PDF: "PDF",
  FALLBACK: "MEDIA"
};

export const getMediaItemMeta = (mediaItem: MediaItem) => {
  if (mediaItem.__typename === "MediaFolder") {
    return META_TYPES.FOLDER;
  }
  if (mediaItem.url) {
    return mediaItem.url.includes("vimeo")
      ? META_TYPES.VIDEO
      : META_TYPES.EXTERNAL_LINK;
  }
  const contentType = mediaItem.media?.contentType;
  if (!contentType) {
    return META_TYPES.FALLBACK;
  }
  if (contentType.includes("image")) {
    return META_TYPES.IMAGE;
  }
  if (contentType.includes("pdf")) {
    return META_TYPES.PDF;
  }
  if (contentType.includes("video")) {
    return META_TYPES.VIDEO;
  }
  return META_TYPES.FALLBACK;
};

export const getMediaItemSize = (mediaItem: MediaItem) => {
  if (mediaItem.__typename !== "MediaTool" || !mediaItem.media) {
    return "";
  }
  const numBytes = formatFileSize(mediaItem.media.size);
  return numBytes;
};

// taken from StackOverflow
// https://stackoverflow.com/a/18650828/5489631
export const formatFileSize = (bytes: number) => {
  if (typeof bytes === "undefined") {
    return "";
  }
  if (bytes === 0) {
    return "0 Bytes";
  }
  const k = 1024;

  const sizes = ["Bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"];

  const i = Math.floor(Math.log(bytes) / Math.log(k));

  return [parseFloat((bytes / Math.pow(k, i)).toFixed(2)), sizes[`${i}`]].join(
    " "
  );
};

export const resizeContentfulImage = (url: string, maxWidth: number) => {
  return `${url}?w=${maxWidth}`;
};

export const getVimeoEmbedUrl = (mediaUrl: MediaTool["url"]) => {
  const videoId = mediaUrl.split("/").slice(-1)[0];
  return `https://player.vimeo.com/video/${videoId}`;
};

export const isVimeo = (mediaItem: MediaItem) => {
  return (
    mediaItem.__typename === "MediaTool" && mediaItem.url?.includes("vimeo")
  );
};

export const isExternalLink = (mediaItem: MediaItem) => {
  return (
    mediaItem.__typename === "MediaTool" &&
    mediaItem.url &&
    !mediaItem.url.includes("vimeo")
  );
};
