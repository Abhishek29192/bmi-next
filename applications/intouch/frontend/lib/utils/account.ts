export const randomPassword = (length: number = 8) => {
  const charset =
    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";

  return [...new Array(length)]
    .map(() => charset.charAt(Math.floor(Math.random() * charset.length)))
    .join("");
};

export const sortByFirstName = (nodes) =>
  [...nodes].sort((a, b) => a.firstName?.localeCompare(b?.firstName));
