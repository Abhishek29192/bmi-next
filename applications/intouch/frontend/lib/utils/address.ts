import { Address } from "@bmi/intouch-api-types";

export const stringifyAddress = (address: Partial<Address>, separator = " ") =>
  [
    address?.firstLine,
    address?.secondLine,
    address?.postcode,
    address?.town,
    address?.country
  ]
    .filter(Boolean)
    .join(separator);

export const copyOfSiteAddress = (address: Partial<Address>) => {
  const { id, __typename, ...parentSiteAddress } = address;
  return parentSiteAddress;
};

export const copyOfBuildingOwnerAddress = (address: Partial<Address>) => {
  const { id, __typename, ...parentBuildingOwnerAddress } = address;
  return parentBuildingOwnerAddress;
};
