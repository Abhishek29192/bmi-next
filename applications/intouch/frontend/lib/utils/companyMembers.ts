export const reorderMembers = (nodes) =>
  [...nodes].sort((a, b) =>
    a.account?.firstName?.localeCompare(b?.account?.firstName)
  );
