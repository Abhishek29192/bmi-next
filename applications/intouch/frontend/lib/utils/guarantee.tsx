import React from "react";
import {
  RequestStatus,
  Technology,
  ContentfulTiers,
  Tier
} from "@bmi/intouch-api-types";
import Check from "@material-ui/icons/Check";
import Warning from "@material-ui/icons/Warning";
import WatchLaterIcon from "@material-ui/icons/WatchLater";
import { GetProjectQuery } from "../../graphql/generated/operations";

export type GuaranteeStatus =
  | Exclude<RequestStatus, "NEW">
  | "READY"
  | "STARTED"
  | "NOT_APPLICABLE";

export const guaranteeStatusIcons: Record<
  GuaranteeStatus,
  React.ReactElement | null
> = {
  NOT_APPLICABLE: null,
  STARTED: <Warning style={{ color: "orange" }} />,
  READY: <Check style={{ color: "green" }} />,
  SUBMITTED: <WatchLaterIcon style={{ color: "blue" }} />,
  REVIEW: <WatchLaterIcon style={{ color: "blue" }} />,
  REJECTED: <Warning style={{ color: "red" }} />,
  APPROVED: <Check style={{ color: "green" }} />
};

export enum SolutionGuaranteeValidationError {
  NoGuarantee,
  NotApplicable,
  MinumumEvidence,
  ResponsibleInstaller,
  Certification,
  Tier,
  BuilderOwner,
  ProjectDetail
}

type SolutionGuaranteeValidateResult = {
  isValid: boolean;
  validationError?: SolutionGuaranteeValidationError;
};

// TODO:
export const guaranteePrerequsitesMet = (guarantee) => true;

export const guaranteeSolutionGuaranteeValidate = (
  project: GetProjectQuery["project"]
): SolutionGuaranteeValidateResult => {
  const guarantee = project?.guarantees?.nodes?.[0];
  if (!guarantee) {
    return {
      isValid: false,
      validationError: SolutionGuaranteeValidationError.NoGuarantee
    };
  }
  if (!["NEW", "REJECTED"].includes(guarantee.status)) {
    return {
      isValid: false,
      validationError: SolutionGuaranteeValidationError.NotApplicable
    };
  }

  const { guaranteeType } = guarantee;
  const { projectMembers, technology, company, evidenceItems } = project;

  const isEvidenceUploaded = checkEvidence(
    evidenceItems?.nodes,
    guaranteeType.evidenceCategoriesCollection.items
  );
  if (!isEvidenceUploaded) {
    //The minimum number or Evidence Items has been uploaded for all the Evidence Categories that apply to the Guarantee Type
    return {
      isValid: false,
      validationError: SolutionGuaranteeValidationError.MinumumEvidence
    };
  }

  const responsibleInsalller = getResponsibleInsalller(projectMembers.nodes);
  if (!responsibleInsalller) {
    //TThere is a Responsible Installer nominated for the Project.
    return {
      isValid: false,
      validationError: SolutionGuaranteeValidationError.ResponsibleInstaller
    };
  }

  const isCertificationAvailable = checkCertification(
    responsibleInsalller,
    technology
  );
  if (!isCertificationAvailable) {
    //The nominated Responsible Installer currently holds a Certification in the same Technology as the Project
    return {
      isValid: false,
      validationError: SolutionGuaranteeValidationError.Certification
    };
  }

  const isTierAvailable = checkCompanyTier(
    guaranteeType.tiersAvailable,
    company.tier
  );
  if (!isTierAvailable) {
    //You are in a Tier that supports the Guarantee Type.
    return {
      isValid: false,
      validationError: SolutionGuaranteeValidationError.Tier
    };
  }

  const isBuilderOwnerCompleted = checkBuildingOwner(project);
  if (!isBuilderOwnerCompleted) {
    //The mandatory Building Owner fields are complete for the Project, i.e.
    return {
      isValid: false,
      validationError: SolutionGuaranteeValidationError.BuilderOwner
    };
  }

  const isProjectDetailsCompleted = checkProjectDetail(project);
  if (!isProjectDetailsCompleted) {
    //The other mandatory Project details are complete present:
    return {
      isValid: false,
      validationError: SolutionGuaranteeValidationError.ProjectDetail
    };
  }
  return {
    isValid: true
  };
};
export const isGuaranteeApplicationEnable = (
  project: GetProjectQuery["project"]
) => {
  if (!checkProjectDetail(project)) {
    return false;
  }

  if (!checkBuildingOwner(project)) {
    return false;
  }

  const { company, guarantees } = project;

  if (guarantees.nodes.length === 0) return true;

  const guarantee = guarantees.nodes[0];
  const { guaranteeType } = guarantee;

  if (!checkCompanyTier(guaranteeType.tiersAvailable, company.tier)) {
    return false;
  }

  return true;
};

const checkEvidence = (
  guaranteEvidenceItems: GetProjectQuery["project"]["evidenceItems"]["nodes"] = [],
  //TODO: Find an alternative way to define parameter type.
  evidenceCategories: GetProjectQuery["project"]["guarantees"]["nodes"][0]["guaranteeType"]["evidenceCategoriesCollection"]["items"]
) => {
  for (const { referenceCode, minimumUploads } of evidenceCategories) {
    const uploadedFileCount = guaranteEvidenceItems.filter(
      (e) => e.customEvidenceCategoryKey === referenceCode
    ).length;

    if (uploadedFileCount < minimumUploads) return false;
  }

  return true;
};

const getResponsibleInsalller = (
  projectMembers: GetProjectQuery["project"]["projectMembers"]["nodes"]
) => {
  return projectMembers.find((member) => member.isResponsibleInstaller);
};

const checkCertification = (
  projectMember: GetProjectQuery["project"]["projectMembers"]["nodes"][0],
  technology: Technology
) => {
  return projectMember.account.certificationsByDoceboUserId.nodes.some(
    (certification) => certification.technology === technology
  );
};

const checkCompanyTier = (availableTiers: ContentfulTiers[], tier: Tier) => {
  return availableTiers.includes(tier);
};

const checkBuildingOwner = ({
  buildingOwnerMail,
  buildingOwnerFirstname,
  buildingOwnerLastname,
  buildingOwnerAddress
}: GetProjectQuery["project"]) => {
  return [
    buildingOwnerMail,
    buildingOwnerFirstname,
    buildingOwnerLastname,
    buildingOwnerAddress
  ].every(Boolean);
};

const checkProjectDetail = ({
  name,
  siteAddress,
  technology,
  roofArea,
  startDate,
  endDate
}: GetProjectQuery["project"]) => {
  return [name, siteAddress, technology, roofArea, startDate, endDate].every(
    Boolean
  );
};
