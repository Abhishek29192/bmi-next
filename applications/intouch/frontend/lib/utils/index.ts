import { Request } from "express";

export const formatDate = (date: string) =>
  new Intl.DateTimeFormat("en-GB", {
    dateStyle: "medium"
  } as any).format(new Date(date?.substring(0, 10)));

type MarketEnvProps = {
  currentHost: string;
  environment: string;
  market: string;
  paths: string[];
};

export const getMarketAndEnvFromReq = ({
  headers
}: Request): MarketEnvProps => {
  let market = null;
  let environment = null;

  let { host } = headers;

  const paths = host.split(".");
  const subdomain = paths[0];

  if (subdomain.indexOf("-") !== -1) {
    const parts = subdomain.split("-");
    environment = parts[0];
    market = parts[1];
  } else {
    market = subdomain;
  }

  return {
    currentHost: host,
    environment,
    market,
    paths
  };
};

export const sortArrayByField = (nodes: any[], field: string) =>
  nodes?.sort((a, b) => a?.[field]?.localeCompare(b?.[field]));
