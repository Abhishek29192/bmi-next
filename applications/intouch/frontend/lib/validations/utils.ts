import { get } from "lodash";
import validator from "validator";
import { ValidationResult } from "@bmi/form";

export type ValidateObjectResult = {
  result: boolean;
  missingFields: string[];
};

export const validateObject = (
  object: { [key: string]: any },
  mandatoryFields: string[]
): ValidateObjectResult => {
  const missingFields = mandatoryFields.filter(
    (mandatoryField) => !get(object, mandatoryField)
  );
  return {
    result: missingFields.length === 0,
    missingFields
  };
};

export const validateUrlInput =
  (t) =>
  (val: string): ValidationResult => {
    const isValid = !val || validator.isURL(val);
    return isValid ? null : t("common:error_messages.url_format");
  };

export const validateEmailInput =
  (t) =>
  (val: string): ValidationResult => {
    const isValid = !val || validator.isEmail(val);
    return isValid ? null : t("common:error_messages.email_format");
  };

const PHONE_NUMBER_MIN_LENGTH = 6;
const PHONE_NUMBER_MAX_LENGTH = 16;

export const validatePhoneNumberInput =
  (t) =>
  (val: string): ValidationResult => {
    const isValid =
      !val ||
      (validator.isLength(val, {
        min: PHONE_NUMBER_MIN_LENGTH,
        max: PHONE_NUMBER_MAX_LENGTH
      }) &&
        val.match(/^[\+\d]?(?:[\d-.\s()]*)$/gm)); // eslint-disable-line no-useless-escape
    return isValid ? null : t("common:error_messages.phone_number_format");
  };
