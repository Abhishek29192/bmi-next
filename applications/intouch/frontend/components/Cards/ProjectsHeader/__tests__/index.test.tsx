import React from "react";
import { Technology } from "@bmi/intouch-api-types";
import merge from "lodash/merge";
import {
  fireEvent,
  renderWithUserProvider,
  screen
} from "../../../../lib/tests/utils";
import AccountContextWrapper from "../../../../lib/tests/fixtures/account";
import { GuaranteeStatus } from "../../../../lib/utils/guarantee";
import { generateAccount } from "../../../../lib/tests/factories/account";
import { ProjectsHeader } from "..";

const mockGuaranteeEventHandler = jest.fn();

const defaultProjectHeaderData = {
  title: "Old Brompton Library",
  technology: "FLAT" as Technology,
  projectCode: "PROFLO-d1847",
  projectStatus: "In progress",
  roofArea: 100,
  buildingAddress: {
    firstLine: "8 Old Brompton Road",
    secondLine: "South Kensington",
    town: "London",
    postcode: "SW7 3SS"
  },
  projectDescription: "Lorem ipsum dolor sit amet",
  startDate: "24 Feb 2021",
  endDate: "30 March 2022",
  guaranteeType: "Not requested",
  guaranteeStatus: "NOT_APPLICABLE" as GuaranteeStatus,
  hidden: true
};

export const generateProjectHeaderData = (projectHeaderData = {}) =>
  merge(defaultProjectHeaderData, projectHeaderData);

describe("ProjectsHeader component", () => {
  it("render correctly", () => {
    const { container } = renderWithUserProvider(
      <AccountContextWrapper account={generateAccount({ role: "SUPER_ADMIN" })}>
        <ProjectsHeader {...generateProjectHeaderData()} />
      </AccountContextWrapper>
    );
    expect(container).toMatchSnapshot();
  });
  it("should show guarante event button if the user is super admin", () => {
    renderWithUserProvider(
      <AccountContextWrapper account={generateAccount({ role: "SUPER_ADMIN" })}>
        <ProjectsHeader
          {...generateProjectHeaderData()}
          guaranteeEventType={"ASSIGN_SOLUTION"}
        />
      </AccountContextWrapper>
    );
    expect(screen.getByTestId("guarantee-event-button")).toBeTruthy();
  });
  it("should show guarante event button if the user is market admin", () => {
    renderWithUserProvider(
      <AccountContextWrapper
        account={generateAccount({ role: "MARKET_ADMIN" })}
      >
        <ProjectsHeader
          {...generateProjectHeaderData()}
          guaranteeEventType={"ASSIGN_SOLUTION"}
        />
      </AccountContextWrapper>
    );
    expect(screen.getByTestId("guarantee-event-button")).toBeTruthy();
  });

  it("should not show guarante event button if the user is installer", () => {
    renderWithUserProvider(
      <AccountContextWrapper account={generateAccount({ role: "INSTALLER" })}>
        <ProjectsHeader
          {...generateProjectHeaderData()}
          guaranteeEventType={"ASSIGN_SOLUTION"}
        />
      </AccountContextWrapper>
    );
    expect(screen.queryByTestId("guarantee-event-button")).toBeFalsy();
  });

  it("should not show archived key/value pair for installer", () => {
    renderWithUserProvider(
      <AccountContextWrapper account={generateAccount({ role: "INSTALLER" })}>
        <ProjectsHeader
          {...generateProjectHeaderData({ hidden: false })}
          guaranteeEventType={"ASSIGN_SOLUTION"}
        />
      </AccountContextWrapper>
    );
    expect(screen.queryByTestId("archived-field")).toBeFalsy();
  });

  it("should show archived key/value pair for SUPER_ADMIN", () => {
    renderWithUserProvider(
      <AccountContextWrapper account={generateAccount({ role: "SUPER_ADMIN" })}>
        <ProjectsHeader
          {...generateProjectHeaderData({
            hidden: null,
            renderActions: () => {},
            guaranteeEventHandler: mockGuaranteeEventHandler
          })}
          guaranteeEventType={"ASSIGN_SOLUTION"}
        />
      </AccountContextWrapper>
    );
    expect(screen.queryByText("projectDetails.archived")).toBeTruthy();
    const guaranteeButton = screen.getByTestId("guarantee-event-button");
    fireEvent.click(guaranteeButton);
    expect(mockGuaranteeEventHandler).toBeCalled();
  });
});
