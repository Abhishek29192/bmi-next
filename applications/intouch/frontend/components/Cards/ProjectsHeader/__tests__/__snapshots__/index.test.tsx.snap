// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsHeader component render correctly 1`] = `
<div
  class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiCardContent-root"
  >
    <div
      style="display: flex;"
    >
      <h3
        class="MuiTypography-root Typography Typography--underline MuiTypography-h3"
      >
        Old Brompton Library
      </h3>
      <svg
        class="technologyIcon"
      />
    </div>
    <div
      class="body"
    >
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.code
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          PROFLO-d1847
        </div>
      </div>
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.status
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          In progress
        </div>
      </div>
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.roofArea
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          100
          m
          <sup>
            2
          </sup>
        </div>
      </div>
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.buildingAddress
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          <div
            data-testid="address"
          >
            <div
              data-testid="addressLine-firstLine"
            >
              8 Old Brompton Road
            </div>
            <div
              data-testid="addressLine-secondLine"
            >
              South Kensington
            </div>
            <div
              data-testid="addressLine-postcode"
            >
              SW7 3SS
            </div>
            <div
              data-testid="addressLine-town"
            >
              London
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.description
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          Lorem ipsum dolor sit amet
        </div>
      </div>
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.startDate
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          February 24, 2021
        </div>
      </div>
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.endDate
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          March 30, 2022
        </div>
      </div>
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.guaranteeType
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          Not requested
        </div>
      </div>
      <div
        style="margin-bottom: 1.5rem;"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          projectDetails.guaranteeStatus
        </h6>
        <div
          class="MuiTypography-root Typography MuiTypography-body1"
        >
          <div
            style="display: flex;"
          >
            guarantee.status.NOT_APPLICABLE
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
